__tests__/components/AnonymousChat/child/ChatItemMyTextV2.test.tsx

```diff
@@ -7,172 +8,98 @@ import {act, fireEvent, render} from '@testing-library/react-native';
 
 import ChatItemMyTextV2 from '../../../../src/components/AnonymousChat/child/ChatItemMyTextV2';
 import {Context} from '../../../../src/context';
-import {DEFAULT_PROFILE_PIC_PATH} from '../../../../src/utils/constants';
+import {DEFAULT_PROFILE_PIC_PATH, MESSAGE_TYPE_REGULAR} from '../../../../src/utils/constants';
+import {SIGNED} from '../../../../src/hooks/core/constant';
+
+export let setReplyPreview;
+export let contextValue;
+export const timestamp = new Date().toISOString();
+
+export const setupTests = () => {
+  setReplyPreview = jest.fn();
+  contextValue = {
+    chat: [{replyTarget: null}, setReplyPreview],
+    profile: [{myProfile: {}}]
+  };
+};
 
-jest.spyOn(React, 'useRef').mockReturnValue({current: {close: jest.fn()}});
-jest.mock('react-native-reanimated', () => ({
-  ...jest.requireActual('react-native-reanimated'),
-  useSharedValue: jest.fn(() => ({value: 0})),
-  useAnimatedStyle: jest.fn(() => ({})),
-  View: require('react-native/Libraries/Components/View/View').View,
-  Animated: {
-    View: require('react-native/Libraries/Components/View/View').View
-  }
-}));
-jest.mock('react-native-gesture-handler', () => {
-  const originalModule = jest.requireActual('react-native-gesture-handler');
-  const {View} = require('react-native');
-  const React = require('react');
+export const renderWithProvider = (Component, props) => {
+  return render(
+    <Context.Provider value={contextValue}>
+      <Component {...props} />
+    </Context.Provider>
+  );
+};
+
+export const expectedReplyPreview = {
+  payload: {
+    id: 'id986',
+    message: 'message',
+    message_type: MESSAGE_TYPE_REGULAR,
+    chatType: SIGNED,
+    updated_at: timestamp,
+    user: {username: 'username'}
+  },
+  type: 'SET_REPLY_TARGET'
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code has been refactored to make it more modular and maintainable. The setup for tests is now in a separate function `setupTests`, and the rendering of the component with the provider is also in a separate function `renderWithProvider`. This makes the test cases cleaner and easier to read. However, I noticed that the `contextValue` object is exported and mutated directly in `setupTests`. It's generally not a good practice to mutate exported objects as it can lead to unexpected side effects. Instead, consider returning a new instance of `contextValue` from `setupTests`.

```diff
- export let contextValue;
+ let contextValue;

export const setupTests = () => {
  setReplyPreview = jest.fn();
- contextValue = {
+ return {
    chat: [{replyTarget: null}, setReplyPreview],
    profile: [{myProfile: {}}]
  };
};

// Then in your beforeEach block
beforeEach(() => {
- setupTests();
+ contextValue = setupTests();
});
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->